src/hydration/tests/useHydrate.test.js

```diff
@@ -0,0 +1,179 @@
+import React from 'react'
+import { render, waitFor } from '@testing-library/react'
+import {
+  ReactQueryCacheProvider,
+  makeQueryCache,
+  useQuery,
+} from '../../react/index'
+import { dehydrate, useHydrate } from '../index'
+
+describe('hydration', () => {
+  function Hydrator({ initialQueries, children }) {
+    useHydrate(initialQueries)
+    return children
+  }
+
+  const fetchData = value =>
+    new Promise(res => setTimeout(() => res(value), 10))
+  const dataQuery = key => fetchData(key)
+  let stringifiedQueries
+
+  beforeAll(async () => {
+    const serverQueryCache = makeQueryCache()
+    await serverQueryCache.prefetchQuery('string', dataQuery)
+    const dehydrated = dehydrate(serverQueryCache)
+    stringifiedQueries = JSON.stringify(dehydrated)
+    serverQueryCache.clear({ notify: false })
+  })
+
+  test('should handle global cache case', async () => {
```

Comment:
I'm not sure how to handle this case properly. Right now it works on the client, but kind of fails silently on the server (since queries wont actually get added to the cache there). Using the global cache with hydration could be nice in some cases with custom SSR with a separate client entry-point, but is a footgun in Next.

Since v3 will remove the global cache, maybe this behavior should be changed to throw an error if no custom cache has been provided via a `ReactQueryCacheProvider` instead?